beancount/parser/lexer.l

```diff
@@ -166,6 +166,7 @@ int pyfile_read_into(PyObject *file, char *buf, size_t max_size);
         yylloc->last_column = yycolumn + yyleng - 1;                    \
         yylloc->file_name = yy_filename;                                \
         yycolumn += yyleng;                                             \
+        if (YY_START == AFTERDATE) BEGIN(INITIAL);                      \
```

Comment:
The plan is to move the lexer to RE/flex where `YY_USER_ACTION` should go away. I would avoid to add more to it. Also, having a state transition in it does not make the mechanism really obvious. See below.